diff --git a/arch/arm/dts/myb-rzg2l-base.dts b/arch/arm/dts/myb-rzg2l-base.dts
index 66bb9081..3836803d 100644
--- a/arch/arm/dts/myb-rzg2l-base.dts
+++ b/arch/arm/dts/myb-rzg2l-base.dts
@@ -27,6 +27,7 @@
 };
 
 &pinctrl {
+/*
 	eth0_pins: eth0 {
 		pinmux = <RZG2L_PINMUX(20, 0, 1)>,
 				<RZG2L_PINMUX(20, 1, 1)>,
@@ -47,7 +48,7 @@
 				<RZG2L_PINMUX(27, 1, 1)>,
 				<RZG2L_PINMUX(28, 0, 1)>;
 	};
-
+*/
 	i2c1_pins: i2c1 {
 		pinmux = <RZG2L_PINMUX(14, 2, 1)>,
 				<RZG2L_PINMUX(14, 3, 1)>;
@@ -108,7 +109,7 @@
 	bus-width = <4>;
 	status = "okay";
 };
-
+/*
 &eth0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&eth0_pins>;
@@ -120,7 +121,7 @@
 		reg = <4>;
 	};
 };
-
+*/
 &i2c1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c1_pins>;
diff --git a/board/myir/myc-rzg2l/myc-rzg2l.c b/board/myir/myc-rzg2l/myc-rzg2l.c
index 9cb9f363..4c79516b 100755
--- a/board/myir/myc-rzg2l/myc-rzg2l.c
+++ b/board/myir/myc-rzg2l/myc-rzg2l.c
@@ -62,6 +62,12 @@ DECLARE_GLOBAL_DATA_PTR;
 #define PFC_PM3B                              (PFC_BASE + 0x176)
 #define PFC_PMC3B                             (PFC_BASE + 0x23B)
 
+/*P26_1*/
+#define PFC_P2A                               (PFC_BASE + 0x02A)
+#define PFC_PM2A                              (PFC_BASE + 0x154)
+#define PFC_PMC2A                             (PFC_BASE + 0x22A)
+
+
 static void board_usb_init(void);
 
 void s_init(void)
@@ -129,8 +135,14 @@ int board_init(void)
 	*(volatile u32 *)(PFC_P3B) = (*(volatile u32 *)(PFC_P3B) & 0xFFFFFFF7) | 0x0;
 	mdelay(13);
 	*(volatile u32 *)(PFC_P3B) = (*(volatile u32 *)(PFC_P3B) & 0xFFFFFFF7) | 0x08;
+
 	
 	/* PINCTRL, USB-PHY, USB_BLK init */
+	*(volatile u32 *)(PFC_PMC2A) &= 0xFFFFFFFC; /* Port func mode  */
+	*(volatile u32 *)(PFC_PM2A) = (*(volatile u32 *)(PFC_PM2A) & 0xFFFFFFCF) | 0x20; /* Port output mode 0b10 */
+
+	*(volatile u32 *)(PFC_P2A) = (*(volatile u32 *)(PFC_P2A) & 0xFFFFFFFD) | 0x02;
+
 	board_usb_init();
 
 	return 0;
diff --git a/configs/myc-rzg2l_ddr1gb_defconfig b/configs/myc-rzg2l_ddr1gb_defconfig
index 7939a320..f9df268f 100644
--- a/configs/myc-rzg2l_ddr1gb_defconfig
+++ b/configs/myc-rzg2l_ddr1gb_defconfig
@@ -32,6 +32,7 @@ CONFIG_CMD_EXT4=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
+CONFIG_FS_CRAMFS=y
 CONFIG_OF_CONTROL=y
 CONFIG_OF_LIST="myc-rzg2l"
 CONFIG_MULTI_DTB_FIT_LZO=y
diff --git a/configs/myc-rzg2l_defconfig b/configs/myc-rzg2l_defconfig
index ca66d2df..5c574a10 100644
--- a/configs/myc-rzg2l_defconfig
+++ b/configs/myc-rzg2l_defconfig
@@ -32,6 +32,7 @@ CONFIG_CMD_EXT4=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
+CONFIG_FS_CRAMFS=y
 CONFIG_OF_CONTROL=y
 CONFIG_OF_LIST="myc-rzg2l"
 CONFIG_MULTI_DTB_FIT_LZO=y
